stellargraph/connector/neo4j/graph.py

```diff
@@ -0,0 +1,97 @@
+__all__ = ["Neo4jStellarGraph"]
+
+import numpy as np
+import scipy.sparse as sps
+from ...core.experimental import experimental
+
+
+@experimental(reason="the class is not fully tested and lacks documentation")
+class Neo4jStellarGraph:
+    def __init__(self, graph_db, is_directed=False):
+
+        self.graph_db = graph_db
+        self._is_directed = is_directed
+
+    def nodes(self):
+        node_ids_query = f"""    
+            MATCH (n)
+            RETURN collect(n.ID) as node_ids
+            """
+        result = self.graph_db.run(node_ids_query)
+
+        return result.data()[0]["node_ids"]
+
+    def node_features(self, node_ids):
+        feature_query = f"""
+            UNWIND $node_id_list AS node_id
+            MATCH(node) WHERE node.ID = node_id
+            RETURN collect(node.features) as features
+            """
+        result = self.graph_db.run(
+            feature_query, parameters={"node_id_list": node_ids},
+        )
+
+        return np.array(result.data()[0]["features"])
+
+    def to_adjacency_matrix(self, node_ids):
+        subgraph_query = f"""
+            UNWIND $node_id_list AS node_id
+
+            // for each node id in every row, collect the random list of its neighbors.
+            CALL apoc.cypher.run(
+
+                'MATCH(cur_node) WHERE cur_node.ID = $node_id
+
+                // find the neighbors
+                MATCH (cur_node)-->(neighbors)
+                WITH collect(neighbors.ID) AS neigh_ids
+                RETURN neigh_ids',
+                {{node_id: node_id}}) YIELD value
+
+            RETURN collect(value.neigh_ids) as neighbors
+            """
+        result = self.graph_db.run(
+            subgraph_query, parameters={"node_id_list": node_ids}
+        )
+        adj_list = result.data()[0]["neighbors"]
+        index = dict(zip(node_ids, range(len(node_ids))))
+
+        def _remove_invalid(arr):
+            return arr[arr != -1]
+
+        def _numpy_indexer(arr):
+            return np.array([index.get(x, -1) for x in arr])
+
+        adj_list = [_remove_invalid(_numpy_indexer(neighs)) for neighs in adj_list]
+
+        indptr = np.cumsum([0] + [len(neighs) for neighs in adj_list])
+        indices = np.concatenate(adj_list)
+        data = np.ones(len(indices), dtype=np.float32)
+        shape = (len(node_ids), len(node_ids))
+        adj = sps.csr_matrix((data, indices, indptr), shape=shape)
+
+        if not self.is_directed() and len(data) > 0:
```

Comment:
if this code remains, can it be shared from StellarGraph rather than being duplicated?

Reply:
yeah I think this can be shared, but it'll need some work to create an `AbstractStellarGraph`